src/main/java/uk/gov/justice/hmpps/prison/service/ImageService.java

```diff
@@ -47,5 +51,41 @@ public ImageDetail findImageDetail(final Long imageId) {
             .map(i -> fullSizeImage ? i.getFullSizeImage() : i.getThumbnailImage());
     }
 
+    @PreAuthorize("hasRole('SYSTEM_USER')")
+    @HasWriteScope
+    @Transactional
+    public ImageDetail putImageForOffender(final String offenderNumber, final boolean fullSizeImage,  final String imageData) {
+        // Check that the offender number exists
+        List<Offender> offenderList = offenderRepository.findByNomsId(offenderNumber);
+        if (offenderList.isEmpty()) {
+            throw EntityNotFoundException.withId(offenderNumber);
+        }
+
+        // Check that there is a booking present and disallow when not present. Must there be a booking?
+        var latestBooking = offenderList.get(0).getLatestBooking();
+        if (latestBooking.isEmpty()) {
+            throw EntityNotFoundException.withMessage("There are no bookings for {}", offenderNumber);
+        }
 
+        // Uses the sequence OFFENDER_IMAGE_ID to get the next value for the ID
+        var newImage = OffenderImage
+            .builder()
+            .captureDateTime(LocalDateTime.now())
+            .orientationType("FRONT")
+            .viewType("FACE")
+            .imageType("OFF_BKG")
+            .active(true)
+            .sourceCode("GEN")
+            .offenderBooking(latestBooking.get())
+            .thumbnailImage(fullSizeImage ? null : Base64.getDecoder().decode(imageData))
+            .fullSizeImage(fullSizeImage ? Base64.getDecoder().decode(imageData) : null)
+            .build();
+
+        final OffenderImage savedImage = offenderImageRepository.save(newImage);
```

Comment:
Approach is often now to use JPA `addImage` on the OffenderBooking record ```    @OneToMany(mappedBy = "offenderBooking", cascade = CascadeType.ALL)
    @Default
    @Exclude
    @BatchSize(size = 25)
    private List<OffenderImage> images = new ArrayList<>();```

Reply:
Altered to use this method on the booking entity.
`public ImageDetail addImage(OffenderImage image) {
        getImages().add(image);
        image.setOffenderBooking(this);
        return image.transform();
  }`

Reply:
This does insert the image correctly, but does not populate the imageId for the response back as no specific `saveAndFlush()`. Would be nice to get the imageId for the response.